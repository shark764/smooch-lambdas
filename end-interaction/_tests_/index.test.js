const axios = require('axios');
const uuidv1 = require('uuid/v1');

jest.mock('axios');
jest.mock('uuid/v1');

uuidv1.mockImplementation(() => '7534c040-534d-11ea-8aa0-c32d6a748e46');

global.process.env = {
  AWS_REGION: 'us-east-1',
  ENVIRONMENT: 'dev',
  DOMAIN: 'domain',
};

const event = {
  Records: [{
    body: JSON.stringify({
      appId: '5e31c81640a22c000f5d7f28',
      interactionId: '667802d8-2260-436c-958a-2ee0f71f73f0',
      tenantId: '66d83870-30df-4a3b-8801-59edff162034',
    }),
  }],
};

const mockGetSecretValue = jest.fn(() => {})
  .mockImplementation(() => ({
    promise: () => ({
      SecretString: JSON.stringify({
        username: 'username',
        password: 'password',
      }),
    }),
  }));

jest.mock('aws-sdk', () => ({
  SecretsManager: jest.fn().mockImplementation(() => ({ getSecretValue: mockGetSecretValue })),
}));

axios.mockImplementation(() => ({
  data: {
    source: 'smooch',
    interruptType: 'customer-disconnect',
    interrupt: {},
  },
}));

const { handler } = require('../index');

describe('end-interaction', () => {
  describe('Everthing is successful', () => {
    beforeAll(async () => {
      await handler(event);
    });
    it('passes in the correct arguments to mockGetSecretValue', async () => {
      expect(mockGetSecretValue.mock.calls).toMatchSnapshot();
    });

    it('passes in the correct arguments to axios', async () => {
      expect(axios.mock.calls).toMatchSnapshot();
    });
  });

  it('throws an error when there is a problem retrieving cx credentials', async () => {
    try {
      mockGetSecretValue.mockRejectedValueOnce(new Error());
      await handler(event);
    } catch (error) {
      expect(Promise.reject(new Error('Error retrieving cx credentials'))).rejects.toThrowErrorMatchingSnapshot();
    }
  });

  it('throws an error when there is problem ending the interaction', async () => {
    try {
      axios.mockRejectedValueOnce(new Error());
      await handler(event);
    } catch (error) {
      expect(Promise.reject(new Error('Error ending the interaction'))).rejects.toThrowErrorMatchingSnapshot();
    }
  });
});
