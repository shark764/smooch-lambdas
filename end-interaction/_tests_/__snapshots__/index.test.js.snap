// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`end-interaction Everthing is successful passes in the correct arguments to axios 1`] = `
Array [
  Array [
    Object {
      "auth": Object {
        "password": "password",
        "username": "username",
      },
      "data": Object {
        "interrupt": Object {},
        "interruptType": "customer-disconnect",
        "source": "smooch",
      },
      "method": "post",
      "url": "https://us-east-1-dev-edge.domain/v1/tenants/66d83870-30df-4a3b-8801-59edff162034/interactions/667802d8-2260-436c-958a-2ee0f71f73f0/interrupts?id=7534c040-534d-11ea-8aa0-c32d6a748e46",
    },
  ],
]
`;

exports[`end-interaction Everthing is successful passes in the correct arguments to mockGetSecretValue 1`] = `
Array [
  Array [
    Object {
      "SecretId": "us-east-1-dev-smooch-cx",
    },
  ],
]
`;

exports[`end-interaction throws an error when there is a problem retrieving cx credentials 1`] = `"Error retrieving cx credentials"`;

exports[`end-interaction throws an error when there is problem ending the interaction 1`] = `"Error ending the interaction"`;
